File: 2566. Maximum Difference by Remapping a Digit/solution.cpp
class Solution {
public:
  int minMaxDifference(int num) {
    char nonZero;
    char nonNine;

    string stringNum = to_string(num);
    for (int index = 0; index < stringNum.size(); index++){
      char numValue = stringNum[index];
      if (numValue != '0'){
        nonZero = numValue;
        break;
      }
    }

    for (int index = 0; index < stringNum.size(); index++){
      char numValue = stringNum[index];
      if (numValue != '9'){
        nonNine = numValue;
        break;
      }
    }

    string minNum = "";
    string maxNum = "";
    for (int index = 0; index < stringNum.size(); index++){
      char numValue = stringNum[index];
      if (numValue == nonZero){
        minNum += "0";
      }
      else{
        minNum += numValue;
      }
      if (numValue == nonNine){
        maxNum += "9";
      }
      else{
        maxNum += numValue;
      }
    }
    
    return stoi(maxNum) - stoi(minNum);
  }
};

File: 2566. Maximum Difference by Remapping a Digit/solution.py
class Solution(object):
  def minMaxDifference(self, num):
    """
    :type num: int
    :rtype: int
    """
    nonZero = ""
    nonNine = ""

    stringNum = str(num)
    for numValue in stringNum:
      if numValue != "0":
        nonZero = numValue
        break

    for numValue in stringNum:
      if numValue != "9":
        nonNine = numValue
        break

    minNum = ""
    maxNum = ""
    for numValue in stringNum:
      if numValue == nonZero:
        minNum += "0"
      else:
        minNum += numValue
      if numValue == nonNine:
        maxNum += "9"
      else:
        maxNum += numValue
    
    return int(maxNum) - int(minNum)